site/documentation/references/security/index.md

```diff
@@ -0,0 +1,288 @@
+```
+
+#### 4. Access the API with HTTPS
+
+```bash
+curl -k https://localhost:8181/v1/data
+```
+
+> We have to use cURL's `-k/--insecure` flag because we are using a
+> self-signed certificate.
+{: .opa-tip }
+
+### Authentication & Authorization
+
+This section shows how to configure OPA to authenticate and authorize client
+requests. Client-side authentication of the OPA API endpoint should be handled
+with TLS.
+
+Authentication and authorization allow OPA to:
+
+- Verify client identities.
+- Control client access to APIs and data.
+
+Both are configured via command line flags:
+
+- ``--authentication=<scheme>`` specifies the authentication scheme to use.
+- ``--authorization=<scheme>`` specifies the authorization scheme to use.
+
+By default, OPA does not perform authentication or authorization and these flags
+default to `off`.
+
+For authentication, OPA supports:
+
+- [Bearer tokens](/documentation/references/rest#bearer-tokens): Bearer tokens are enabled by starting OPA with ``--authentication=token``.
+
+For authorization, OPA relies on policy written in Rego. Authorization is
+enabled by starting OPA with ``--authorization=basic``.
+
+When the `basic` authorization scheme is enabled, a minimal authorization policy
+must be provided on startup. The authorization policy must be structured as follows:
+
+```ruby
+# The "system" namespace is reserved for internal use
+# by OPA. Authorization policy must be defined under
+# system.authz as follows:
+package system.authz
+
+default allow = false  # Reject requests by default.
+
+allow {
+  # Logic to authorize request goes here.
+}
+```
+
+When OPA receives a request, it executes a query against the document defined
+`data.system.authz.allow`. The implementation of the policy may span multiple
+packages however it is recommended that administrators keep the policy under the
+`system` namespace.
+
+If the document produced by the ``allow`` rule is ``true``, the request is
+processed normally. If the document is undefined or **not** ``true``, the
+request is rejected immediately.
+
+OPA provides the following input document when executing the authorization
+policy:
+
+```ruby
+{
+    "input": {
+        # Identity established by authentication scheme.
+        # When Bearer tokens are used, the identity is
+        # set to the Bearer token value.
+        "identity": <String>,
+
+        # One of {GET, POST, PUT, PATCH, DELETE}.
+        "method": <HTTP Method>,
+
+        # URL path represented as an array.
+        # For example: /v1/data/exempli-gratia
+        # is represented as [v1, data, exampli-gratia]
+        "path": <HTTP URL Path>,
+    }
+}
+```
+{: .opa-collapse--ignore}
+
+At a minimum, the authorization policy should grant access to a special root
+identity:
+
+```ruby
+package system.authz
+
+default allow = false          # Reject requests by default.
+
+allow {                        # Allow request if...
+    input.identity = "secret"  # Identity is the secret root key.
```

Comment:
To mitigate potential confusion over “assignment”, perhaps: `"secret" = input.identity`?